### PR TITLE
Charts API

### PR DESCRIPTION
**Pull request by ChenyuHeidiZhang**
Created 2 charts - one personal timeline chart on index.html, one comments mood pie chart on comments.html.

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/CommentsMoodServlet.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that fetches counts for different types of comment mood. */
```
Comment (hayleejane3) [inline portfolio/src/main/java/com/google/sps/servlets/CommentsMoodServlet.java]: I would explain things a little better here. 
"counts for different types of comment mood" doesn't give away much so someone who has no context wouldn't understand what's going on.

File: portfolio/src/main/webapp/script-timeline-chart.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+google.charts.load('current', {packages:['timeline']});
+google.charts.setOnLoadCallback(drawTimelineChart);
+
+/** 
+ * Draws the timeline chart onto the DOM.
```
Comment (hayleejane3) [inline portfolio/src/main/webapp/script-timeline-chart.js]: Context on what the timeline is about?

File: portfolio/src/main/webapp/script-timeline-chart.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+ */
```
Comment (hayleejane3) [inline portfolio/src/main/webapp/script-timeline-chart.js]: You can make this a single line comment with the same  `/**` format

File: portfolio/src/main/webapp/script.js [hunk 4]
```diff
@@ -13,11 +13,14 @@
 // limitations under the License.
 
 /**
- * Toggles the drop down text.
+ * Toggles the dropdown text and dropdown arrow.
  * @param {string} id The id of the dropdown content to be toggled.
+ * @param {!Event} event The click event on a dropdown bar.
```
Comment (hayleejane3) [inline portfolio/src/main/webapp/script.js:18]:
```suggestion
 * @param {!Event} event The click event for the dropdown bar.
```

Comment (hayleejane3) [reply portfolio/src/main/webapp/script.js:18]: Should it be dropdown content?

File: portfolio/src/main/webapp/script.js [hunk 5]
```diff
@@ -13,11 +13,14 @@
...
  */
-function toggleDropDown(id) {
-  document.getElementById(id).classList.toggle("show");
+function toggleDropDown(id, event) {
+  document.getElementById(id).classList.toggle('show');
```
Comment (hayleejane3) [inline portfolio/src/main/webapp/script.js:21]: Shouldn't you. be assigning this to some variable?

Comment (hayleejane3) [reply portfolio/src/main/webapp/script.js:21]: Or is this leftover code.

File: portfolio/src/main/webapp/style.css [hunk 6]
```diff
@@ -162,10 +162,19 @@ h3 {
...
   transform: rotate(45deg);
   -webkit-transform: rotate(45deg);
 }
 
+.up {
+  margin-top: 8px;
+  transform: rotate(-135deg);
+  -webkit-transform: rotate(-135deg);
+}
+
```
Comment (hayleejane3) [inline portfolio/src/main/webapp/style.css]: You might want to add comments on what these are for.

File: portfolio/src/main/webapp/style.css [hunk 7]
```diff
@@ -162,10 +162,19 @@ h3 {
...
 /* Dropdown button. */
```
Comment (hayleejane3) [inline portfolio/src/main/webapp/style.css]: Going off of the previous comment^, i guess update this one too.

**Review by hayleejane3 — CHANGES_REQUESTED** [review]

File: portfolio/src/main/java/com/google/sps/servlets/CommentsMoodServlet.java (hunk 1 above)
Comment (ChenyuHeidiZhang) [reply portfolio/src/main/java/com/google/sps/servlets/CommentsMoodServlet.java]: okay

File: portfolio/src/main/webapp/script-timeline-chart.js (hunk 2 above)
Comment (ChenyuHeidiZhang) [reply portfolio/src/main/webapp/script-timeline-chart.js]: ok

File: portfolio/src/main/webapp/script-timeline-chart.js (hunk 3 above)
Comment (ChenyuHeidiZhang) [reply portfolio/src/main/webapp/script-timeline-chart.js]: sure

File: portfolio/src/main/webapp/script.js (hunk 4 above)
Comment (ChenyuHeidiZhang) [reply portfolio/src/main/webapp/script.js:18]: I mean that the click is on the dropdown bar?

File: portfolio/src/main/webapp/script.js (hunk 5 above)
Comment (ChenyuHeidiZhang) [reply portfolio/src/main/webapp/script.js:21]: Assign what? 
So the function is called when the dropdown bar is clicked. The arrow at the right of the bar will be changed from pointing down to pointing up.

File: portfolio/src/main/webapp/style.css (hunk 6 above)
Comment (ChenyuHeidiZhang) [reply portfolio/src/main/webapp/style.css]: okay

File: portfolio/src/main/webapp/style.css (hunk 7 above)
Comment (ChenyuHeidiZhang) [reply portfolio/src/main/webapp/style.css]: got it

File: portfolio/src/main/webapp/script.js (hunk 4 above)
Comment (hayleejane3) [reply portfolio/src/main/webapp/script.js:18]: Yes :)

**Review by hayleejane3 — APPROVED** [review]

**Pull request merged**